src/botPage/view/blockly/blocks/trade/components.js

```diff
@@ -4,65 +4,31 @@ import { symbolApi } from '../../../shared';
 import { oppositesToDropdown } from '../../utils';
 import { caution } from '../images';
 import { getTradeType } from './tools';
+import { getDependentDropdownCallback } from '../shared';
 
 export const marketDropdown = block => {
     const markets = symbolApi.activeSymbols.getMarkets();
-    const getSubmarkets = () => {
-        const marketName = block.getFieldValue('MARKET_LIST');
-        if (marketName === 'Invalid') {
-            return [['', 'Invalid']];
-        }
-        const { submarkets } = markets[marketName];
-        return Object.keys(submarkets).map(e => [submarkets[e].name, e]);
-    };
-    const getSymbols = () => {
-        const submarketName = block.getFieldValue('SUBMARKET_LIST');
-        if (!submarketName || submarketName === 'Invalid') {
-            return [['', '']];
-        }
-        const marketName = block.getFieldValue('MARKET_LIST');
-        const { submarkets } = markets[marketName];
-        const { symbols } = submarkets[submarketName];
-        return Object.keys(symbols).map(e => [symbols[e].display, symbols[e].symbol]);
-    };
     const getMarket = () => Object.keys(markets).map(e => [markets[e].name, e]);
```

Comment:
`getMarket` could be moved with the other list returning functions. Is there a reason for keeping it here?